src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -2875,5 +3025,179 @@ static object G(C c, I<string?> x, I<object?> y)
                 //             x: x] // warn 2
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInArgument, "x").WithArguments("I<string?>", "I<string>", "x", "int C.this[I<string> x, I<object?> y]").WithLocation(15, 16));
         }
+
+        [Fact]
+        public void Conversions_01()
+        {
+            var source =
+@"class A<T> { }
+class B<T> : A<T> { }
+class C
+{
+    static void F(B<object> x)
+    {
+        A<object?> y = x;
+    }
+    static void G(B<object?> x)
+    {
+        A<object> y = x;
+    }
+    static void H(B<object>? x)
+    {
+        A<object?> y = x;
```

Comment:
>A<object?> y = x; [](start = 8, length = 17)

I would also expect a warning about possible null reference assignment on this line.